src/js/components/DateInput/utils.js

```diff
@@ -125,6 +125,30 @@ const pullDigits = (text, index) => {
   return text.slice(index, end);
 };
 
+export const validateDateBounds = ({ dateBounds, selectedDate }) => {
+  if (!dateBounds || !selectedDate) return selectedDate;
+
+  const [startDate, endDate] = dateBounds.map((date) =>
+    new Date(date).toISOString(),
+  );
+
+  const selectedDateArray = Array.isArray(selectedDate)
+    ? selectedDate
+    : [selectedDate];
+
+  const isoSelectedDateArray = selectedDateArray.map((date) =>
+    new Date(date).toISOString(),
+  );
+
+  const allDatesOk = isoSelectedDateArray.every(
+    (isoSelectedDate) =>
+      (!endDate && startDate === isoSelectedDate) ||
```

Comment:
What is this first condition checking for? `!endDate && startDate === isoSelectedDate` It seems like the second condition should be enough .

Reply:
oh this is because the `calendarProps: { bounds: ` can be a array with just one item, and when this happens the only date that can be selected is the same date as in bounds. So if I don't check for this the value will always be false.

Reply:
Thank you for the clarification.